bot/plugins/instance.py

```diff
@@ -395,41 +469,69 @@ async def execute(self) -> None:
             await plugin.app.rest.trigger_typing(self.channel)
 
         # try to execute code
-        att: hikari.Bytes | None = None
-        out: str | None
+        code_file: hikari.Bytes | None = None
+        stdin_file: hikari.Bytes | None = None
+        out: list[str] = [f"<@{self.requester}>"]
+
+        code_output_in_file = False
+        stdin_in_file = False
+
+        code_output: str = ""
         if self.runtime:
             ret = await self.runtime.provider.execute(self)
-            out = ret.format()
+            code_output = ret.format()
 
-            if len(out) > 1_950:
-                att = hikari.Bytes(out, "output.ansi")
-                out = None
-            else:
-                if out in {"", "\n"}:
-                    out = "Your code ran with no output."
-                else:
-                    out = f"```ansi\n{out}\n```"
+            if not code_output.strip():
+                out.append("Your code ran with no output.")
+                code_output = ""
         else:
-            out = "No runtime selected."
+            out.append("No runtime selected.")
 
-        if out:
-            out = f"<@{self.requester}>\n{out}"
-        else:
-            out = f"<@{self.requester}>"
+        stdin = self.stdin or ""
+        formatted_stdin = "\n".join(
+            f"\x1b[1;33mIN:\x1b[0m {line}" for line in stdin.splitlines()
+        )
+
+        if len(code_output) + len(formatted_stdin) > 1_950:
+            if len(code_output) < 1_950:
+                stdin_in_file = True
+            elif len(formatted_stdin) < 1_950:
+                code_output_in_file = True
+            else:
+                code_output_in_file = True
+                stdin_in_file = True
+
+        if code_output:
+            if code_output_in_file:
+                code_file = hikari.Bytes(code_output, "code.ansi")
+            else:
+                out.append(f"```ansi\n{code_output}```")
+
+        if stdin:
+            if stdin_in_file:
+                stdin_file = hikari.Bytes(stdin, "stdin.txt")
+            else:
+                out += [f"```ansi\n{formatted_stdin}```"]
```

Comment:
```suggestion
                out.append(f"```ansi\n{formatted_stdin}```")
```

Reply:
since you merged the other suggestions I went ahead and just added this